src/test/java/uk/gov/pay/adminusers/service/ServiceUpdaterTest.java

```diff
@@ -276,4 +272,94 @@ public void shouldUpdateCurrentGoLIveStageSuccessfully() {
+                getServiceUpdateRequest(addressLine1, "merchant_details/address_line_1"),
+                getServiceUpdateRequest(addressLine2, "merchant_details/address_line_2"),
+                getServiceUpdateRequest(addressCity, "merchant_details/address_city"),
+                getServiceUpdateRequest(addressCountry, "merchant_details/address_country"),
+                getServiceUpdateRequest(addressPostcode, "merchant_details/address_postcode"),
+                getServiceUpdateRequest(email, "merchant_details/email"),
+                getServiceUpdateRequest(telephoneNumber, "merchant_details/telephone_number")
+        );
+        ServiceEntity serviceEntity = ServiceEntityBuilder
+                .aServiceEntity()
+                .withMerchantDetailsEntity(null)
+                .build();
+
+        when(serviceDao.findByExternalId(serviceId)).thenReturn(Optional.of(serviceEntity));
+
+        Optional<Service> maybeService = updater.doUpdate(serviceId, serviceUpdateRequests);
+
+        assertThat(maybeService.isPresent(), is(true));
+        verify(serviceDao, times(8)).merge(serviceEntity);
+        assertThat(maybeService.get().getMerchantDetails().getName(), is(name));
+        assertThat(maybeService.get().getMerchantDetails().getAddressLine1(), is(addressLine1));
+        assertThat(maybeService.get().getMerchantDetails().getAddressLine2(), is(addressLine2));
+        assertThat(maybeService.get().getMerchantDetails().getAddressCity(), is(addressCity));
+        assertThat(maybeService.get().getMerchantDetails().getAddressCountry(), is(addressCountry));
+        assertThat(maybeService.get().getMerchantDetails().getAddressPostcode(), is(addressPostcode));
+        assertThat(maybeService.get().getMerchantDetails().getEmail(), is(email));
+        assertThat(maybeService.get().getMerchantDetails().getTelephoneNumber(), is(telephoneNumber));
+    }
+
+
+    @Test
+    public void shouldUpdateMerchantDetailsAddressLine1Successfully_WhenExistingMerchantDetails() {
+        String name = "Cake service";
+        String serviceId = randomUuid();
+        String addressLine1 = "1 Spider Lane";
+        ServiceUpdateRequest request = getServiceUpdateRequest(addressLine1, "merchant_details/address_line_1");
+        MerchantDetailsEntity merchantDetails = MerchantDetailsEntityBuilder.aMerchantDetailsEntity().withName(name).build();
+        ServiceEntity serviceEntity = ServiceEntityBuilder.aServiceEntity().withMerchantDetailsEntity(merchantDetails).build();
+
+        when(serviceDao.findByExternalId(serviceId)).thenReturn(Optional.of(serviceEntity));
+
+        Optional<Service> maybeService = updater.doUpdate(serviceId, request);
+
+        assertThat(maybeService.isPresent(), is(true));
+        verify(serviceDao).merge(serviceEntity);
+        assertThat(maybeService.get().getMerchantDetails().getName(), is(name));
+        assertThat(maybeService.get().getMerchantDetails().getAddressLine1(), is(addressLine1));
+    }
```

Comment:
Maybe this test could also assert the value of one of the unchanged fields just to confirm that they don’t get blown away

Reply:
👍 Nice one! I've amended this now, so we use`MerchantDetailsEntityBuilder` default values for merchant details fields. So, now on line: 341 we assert that the default name value is still "test-name" after updating addressLine1